server/controllers/profile.controller.js

```diff
@@ -71,6 +71,68 @@ class ProfileController {
     delete dataValues.password;
     return Response(res, 200, 'Profile updated successfully.', [dataValues]);
   }
+
+  /**
+     * @description follows another user
+     * @param {object} req
+     * @param {object} res
+     * @returns {object} user that has been followed
+     */
+  static async followUser(req, res) {
+    const { username } = req.params;
+    const { id } = req.user;
+    const requestingFollowership = await User.findByPk(id);
+    const personToBeFollowed = await User.findOne({ where: { username } });
+    const successMessage = 'Successfully followed user';
+
+    if (!personToBeFollowed) {
+      return Response(res, 404, 'You cannot follow a user that does not exist');
+    }
+    if (requestingFollowership.id === personToBeFollowed.id) {
+      return Response(res, 409, 'You cannot follow yourself');
+    }
+    const newFollowing = await personToBeFollowed.addFollowers(requestingFollowership);
+    if (!newFollowing) {
+      return Response(res, 400, 'You cannot follow a user more than once');
+    }
+    const followers = await personToBeFollowed.getFollowers();
+    const userData = { personToBeFollowed: personToBeFollowed.username };
+    userData.followers = followers.map(follow => ({
+      follower: follow.username
+    }));
+    return Response(res, 200, successMessage, userData);
```

Comment:
Very good job on the use of pivot tables and the many to many relationship. This just helped to clear up one aspect of sequelize I have been struggling with. However, your inline comment states that the above function returns the user that has been followed. Why then do we have an array of followers in the response? Do you think this information is useful to the user? Or did you add it for the sake of clarity?

Reply:
@TheDrizzyWay Yeah the response data containing the array of followers is to show the user has been added to another users followership list.